packages/main/cypress/specs/DateRangePicker.cy.tsx

```diff
@@ -0,0 +1,570 @@
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("27/09/2019 - 10/10/2019")
+			.realPress("Enter");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.then($datePicker => {
+				cy.wrap({
+					startDateValue: () => $datePicker[0].startDateValue,
+					endDateValue: () => $datePicker[0].endDateValue
+				})
+					.as("dates")
+					.invoke("startDateValue")
+					.then(date => {
+						const timestamp = date ? date.getTime() : new Date().getTime();
+						expect(new Date(timestamp)).to.deep.equal(new Date(2019, 8, 27));
+					});
+
+				cy.get("@dates")
+					.invoke("endDateValue")
+					.then(date => {
+						const timestamp = date ? date.getTime() : new Date().getTime();
+						expect(new Date(timestamp as number)).to.deep.equal(new Date(2019, 9, 10));
+					});
+			});
+	});
+
+	it("Setting the same date for first & last is possible", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("05/08/2020 - 05/08/2020")
+			.realPress("Enter");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.then($el => {
+				cy.wrap({
+					startDateValue: () => $el[0].startDateValue,
+					endDateValue: () => $el[0].endDateValue
+				})
+					.as("dates")
+					.invoke("startDateValue")
+					.then(date => {
+						const timestamp = date ? date.getTime() : new Date().getTime();
+						expect(new Date(timestamp)).to.deep.equal(new Date(2020, 7, 5));
+					});
+
+				cy.get("@dates")
+					.invoke("endDateValue")
+					.then(date => {
+						const timestamp = date ? date.getTime() : new Date().getTime();
+						expect(new Date(timestamp as number)).to.deep.equal(new Date(2020, 7, 5));
+					});
+			});
+	});
+
+	it("Change event fired once", () => {
+		cy.mount(<DateRangePicker></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.shadow()
+			.find("ui5-daypicker")
+			.shadow()
+			.find(".ui5-dp-root .ui5-dp-content div > .ui5-dp-item")
+			.as("items")
+			.eq(5)
+			.as("dayOne");
+
+		cy.get("@items")
+			.eq(15)
+			.as("dayTwo");
+
+		cy.get("@dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realPress("F4");
+
+		cy.get("@dateRangePicker")
+			.then($dateRangePicker => {
+				$dateRangePicker.on("change", cy.stub().as("changeStub"));
+			});
+
+		cy.get("@dayOne").realClick();
+		cy.get("@dayTwo").realClick();
+
+		cy.get("@changeStub").should("be.calledOnce");
+	});
+
+	it("Page up/down increments/decrements the day value", () => {
+		cy.mount(<DateRangePicker
+			formatPattern="MMM d, y"
+			value="Jul 16, 2020 @ Jul 29, 2020"
+			delimiter="@">
+		</DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.as("input")
+			.realClick()
```

Comment:
i think it will be good to ensure that the input is focused after the click and before we do actions